linkerd/protocol/http/src/main/scala/io/buoyant/linkerd/protocol/HttpConfig.scala

```diff
@@ -112,9 +133,24 @@ case class HttpConfig(
   override val protocol: ProtocolInitializer = HttpInitializer
 
   @JsonIgnore
+  private[this] val combinedIdentifier = identifier.map { identifierConfigs =>
+    Http.param.HttpIdentifier { (path, dtab) =>
+      identifierConfigs.map(_.newIdentifier(path, dtab))
+        .reduceLeft { (id0, id1) =>
+          { req =>
+            id0(req).flatMap {
+              case identified: IdentifiedRequest[Request] => Future.value(identified)
+              case unidentified: UnidentifiedRequest[Request] => id1(req)
+            }
+          }
```

Comment:
I think if we just had a `def compositeIdentifiers(id0, id1)`, this might read a lot clearer